src/cpp/dpm_sampler.cpp

```diff
@@ -44,15 +45,59 @@
 #include <nmmintrin.h>
 #endif
 
-#include <gsl/gsl_randist.h>
-#include <gsl/gsl_rng.h>
-#include <gsl/gsl_sf_exp.h>
-#include <gsl/gsl_sf_log.h>
-#include <gsl/gsl_sf_pow_int.h>
-
 #include "data_structures.hpp"
 #include "dpm_sampler.hpp"
 
+#include <random>
+#include <boost/random/beta_distribution.hpp>
+std::mt19937 rg;
+
+
+
+// Simple naive implementation of a discrete distribution, using cumulated weights.
+// For generating a single number, that's enough.
+//
+// (as opposed to gsl's, STL's and boost's implementations that all allocate and compute extra look-up tables
+// to further accelerate multiple futur calls to generator, see:
+//  - https://www.gnu.org/software/gsl/manual/html_node/General-Discrete-Distributions.html
+//  - https://stats.stackexchange.com/a/26868 )
+template<typename UNS, typename CDF_type> inline UNS one_shot_discrete_cdf(CDF_type& cdf, const unsigned n, const double* wa) {
```

Comment:
1. why is this `inline`?
2. why not also in an unnamed namespace?

Reply:
I have a bit ancient C++ style syntax, that I still have kept around because I'm used to users trying to compile the code on very antiquated versions of GCC.

By now (2020 - as we are boosting our work to process SARS-cov-2 data), I hope that even CentOS 6 and 7 users are using bioconda (which defaults to GCC 7) as the prefered installation method.